jaeger-apachehttpclient/src/main/java/com/uber/jaeger/httpclient/TracingInterceptors.java

```diff
@@ -32,13 +32,15 @@
 public class TracingInterceptors {
   public static HttpAsyncClientBuilder addTo(HttpAsyncClientBuilder clientBuilder, Tracer tracer) {
     return clientBuilder
```

Comment:
shouldn't `TracingRequestInterceptor` be also changed in this PR to invoke the two new interceptors? And this class won't need to change.

Reply:
So `TracingRequestInterceptor` is the one that was introducing the coupling. I chose to leave it as is, since there might be existing Jaeger users that are directly using that to instrument their clients. Given that you bring up this point, a change can be made to [`TracingRequestInterceptor.process(..)`](https://github.com/jaegertracing/jaeger-client-java/blob/master/jaeger-apachehttpclient/src/main/java/com/uber/jaeger/httpclient/TracingRequestInterceptor.java#L58), to have instances of `SpanCreationRequestInterceptor` and `SpanInjectionRequestInterceptor`, and then invoking the respective `process` methods serially. This would look something like:

```
public class TracingRequestInterceptor implements HttpRequestInterceptor {
  private final Tracer tracer;
  private final SpanCreationRequestInterceptor spanCreationInterceptor;
  private final SpanInjectionRequestInterceptor spanInjectionInterceptor;

  public TracingRequestInterceptor(Tracer tracer) {
    this.tracer = tracer;
    this.spanCreationInterceptor = new SpanCreationRequestInterceptor(tracer);
    this.spanInjectionInterceptor = new SpanInjectionRequestInterceptor(tracer);
  }

  @Override
  public void process(HttpRequest httpRequest, HttpContext httpContext)
      throws HttpException, IOException {
    this.spanCreationInterceptor.process(httpRequest, httpContext);
    this.spanInjectionInterceptor.process(httpRequest, httpContext);
  }
  ..........
}
``` 


Since `TracingInterceptors` is the guy responsible for giving the "complete package", I made the change in for this to use the new interceptors, and thus, proving complete backwards compatibility.

Reply:
> This would look something like:

Yes, that's what I think we should have. Currently you copy & pasted the same code, instead of refactoring to have a single instance of the actual code.